### ISSUE TITLE
Latest SMC (via FlatPak) crashes system when going to General Settings

### ISSUE THREAD

**Issue by amiga-500** [hakandundar34coding/system-monitoring-center#177]
- Using SMC v 2.14.0 (latest from Flatpak)
- Ubuntu 22.10

Decided to run the flatpak file thru terminal in hopes it would say some error messages.  None.

This problem never occurred before in 22.10.  Seems to only crash my system when i go to "General Settings" in 22.10.  No other dialog e.g. About does this.  Some times it goes back to the Login screen (logged me out) but most times it freezes with the black screen u see in the video.  To be clear, this only occurs in Ubuntu 22.10, not any other version that i tested it on (ie. Ubuntu 20.04, 22.04, and 23.04 work fine).

This doesnt affect me directly since i use 23.04 and 20.04 but figured i should bring this up in case other users of your app using 22.10 come knockin'.  There is a good chance tho anyone using 22.10 will quickly upgrade to a newer Ubuntu (e.g. 23.04+) so most users may not ever notice this issue with your app.  So probably not a big deal to fix.

![sss](https://github.com/hakandundar34coding/system-monitoring-center/assets/5191497/f9127d69-ee33-487c-839f-7b7000ea6236)

via syslog
```
May 24 17:00:11 test gnome-software[9107]: failed to refine io.github.hakandundar34coding.system-monitoring-center: failed to refine app: 'utf-8' codec can't decode byte 0xb0 in position 0: invalid start byte
May 24 17:01:54 test gnome-software[9107]: cannot set 101% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:55 test gnome-software[9107]: cannot set 101% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:55 test gnome-software[9107]: cannot set 102% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:56 test gnome-software[9107]: cannot set 102% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:56 test gnome-software[9107]: cannot set 103% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:57 test gnome-software[9107]: cannot set 103% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:57 test gnome-software[9107]: cannot set 103% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:01:58 test gnome-software[9107]: cannot set 103% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:02:00 test gnome-software[9107]: cannot set 103% for system/flatpak/flathub/io.github.hakandundar34coding.system-monitoring-center/stable, setting instead: 100%
May 24 17:02:05 test gnome-software[9107]: libostree pull from 'flathub' for app/io.github.hakandundar34coding.system-monitoring-center/x86_64/stable complete#012security: GPG: summary+commit #012security: SIGN: disabled http: TLS#012delta: parts: 1 loose: 116#012transfer: secs: 1 size: 4.4 MB
May 24 17:02:05 test gnome-software[9107]: /var/tmp/flatpak-cache-IE2Z41/io.github.hakandundar34coding.system-monitoring-center-IR8941/repo-KfHmHn: Pulled app/io.github.hakandundar34coding.system-monitoring-center/x86_64/stable from flathub
May 24 17:02:06 test flatpak-system-helper[9641]: system: Pulled app/io.github.hakandundar34coding.system-monitoring-center/x86_64/stable from /var/lib/flatpak/repo/tmp/flatpak-cache-8TFA51/repo-KfHmHn
May 24 17:02:06 test flatpak-system-helper[9641]: system: Installed app/io.github.hakandundar34coding.system-monitoring-center/x86_64/stable from flathub
May 24 17:02:08 test gnome-software[9107]: failed to refine io.github.hakandundar34coding.system-monitoring-center: failed to refine app: 'utf-8' codec can't decode byte 0xe5 in position 2: invalid continuation byte
May 24 17:02:08 test gnome-software[9107]: failed to refine io.github.hakandundar34coding.system-monitoring-center: failed to refine app: 'utf-8' codec can't decode byte 0x9f in position 99: invalid start byte
May 24 17:02:13 test gnome-software[9107]: failed to refine io.github.hakandundar34coding.system-monitoring-center: failed to refine app: 'utf-8' codec can't decode byte 0xb8 in position 16: invalid start byte
May 24 17:07:30 test io.github.hakandundar34coding.system-monitoring-center.desktop[2773]: failed to create drawable
```

Note: When the system freeze occurs, after rechecking the logs i see nothng written in the syslogs..So the above stuff appears to occur as SMC is running.

Update:

When downloading your latest source code and testing "General Settings" it also crashes.  So its not a FlatPak issue.  Its either something wrong with my system (now) or the app code.
```shell
> wget https://github.com/hakandundar34coding/system-monitoring-center/archive/refs/heads/master.zip
> unzip master.zip
> cd ./system-monitoring-center-master/src
> python3 ./run_from_source.py

TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
TypeError: Couldn't find foreign struct converter for 'cairo.Context'
```

I should point out...for some reason when running the source code directly, none of the animating graphics found under CPU, Disk, etc show at all.  But i am ignoring that since the FlatPak version does show fine.   The "General Settings" crash is still there in this direct source code running though.

**Comment by hakandundar34coding** [comment]
About the following error:
`TypeError: Couldn't find foreign struct converter for 'cairo.Context'`

Can you install the following dependency, run the application from its source code and write the result?
`python3-gi-cairo`

If the application is installed by using a package manager, it installs the dependencies. But they are not installed when the application is run from source code.

I tested the application on virtual machine. The dependency was installed and there was no error. It may have been removed from your system because of some uninstallation operations.

**Comment by amiga-500** [comment]
> Can you install the following dependency, run the application from its source code and write the result?
python3-gi-cairo

HA!...success!  Thank you!  It runs perfectly from the source now.  Yea, you're spot on about the uninstallation operations.  I totally forgot about that.  When upgrading from 20.04 to 22.10, i deinstalled the source which likely removed python3-gi-cairo too since it had no more apps depending on it, and when i got to 22.10 and manually installed the source again it was just not there.

Thanks again.    Still not sure why 22.10's general settings doesnt work via flatpak but i really doubt anyone will notice this since they will likely be upgrading past it soon after to 23.04 where it does work.

**Comment by hakandundar34coding** [comment]
Are there similar problems when you install the application by using Flatpak on a new installation of v22.10?

**Comment by amiga-500** [comment]
i will have to test it out.

**Comment by amiga-500** [comment]
Ok Confirmed...I did a new install of v22.10 and SMC works perfectly.  So yea, it was just an issue when i was upgrading from 22.04 to 22.10.  So something on my side.  Whoops.  Thx for the tip.

Update:

I figured out what was causing the crash.  It seems in VMWare when u have Accelerated Graphics enabled in 22.10 SMC crashes when going to General Settings.  It doesnt seem to have this problem in any other Ubuntu version with Accelerated Graphics.  So i have to assume its something glitchy about the vmware accelerated driver in that ubuntu version.
